Empire/Campaigns.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Empire;
+
+/**
+ * Manages the ads.txt data and presentation
+ *
+ * @package Empire
+ */
+class Campaigns {
+    /**
+     * @var Empire
+     */
+    private $empire;
+
+    const FIELD_EMPIRE_CAMPAIGN_ASSET_FIELD_KEY = 'field_empire_campaign_asset';
+
+    public function __construct( Empire $empire ) {
+        $this->empire = $empire;
+        add_filter(
+            'acf/load_field/key='.self::FIELD_EMPIRE_CAMPAIGN_ASSET_FIELD_KEY,
+            [ $this, 'loadCampaignAssetChoices' ],
+        );
+    }
+
+    public function loadCampaignAssetChoices( $field ) {
+        $field['choices'] = [];
+
+        $assets = $this->empire->loadCampaignsAssets();
+        foreach ($assets as $asset) {
+            $field['choices'][$asset['guid']] = (
+                "#{$asset['externalId']} {$asset['name']} [{$asset['campaign']['name']}]"
+            );
+        }
+
+        return $field;
+    }
```

Comment:
It should be possible to use Organic WP Plugin with any WP installation not only with SWP and we should not rely on assumptions about how data is stored/saved in WP installation. With this approach we kind of force the client to use ACF fields in their WP.

So, in the case of SWP, this part of the code should go to the SWP codebase *and the plugin should only provide a method to pull assets.* That's what @jdemaris originally suggested [in the EP-837](https://recurrentventures.atlassian.net/browse/EP-837?focusedCommentId=21183):
> add a function (maybe empire_get_all_campaigns) that can fetch a list of campaigns from the Campaigns API (restricted to the current org, of course)

But instead of a global function, I would suggest [to use something like](https://github.com/orgnc/wordpress-plugin/blob/master/Empire/Empire.php#L233):
```php
Empire\Empire::getInstance()->getAllCampaigns()
```
What do you think @rrdial?

Reply:
Thanks for the review, @sergunich 👍 

> But instead of a global function, I would suggest to use something like:
```php
Empire\Empire::getInstance()->getAllCampaigns()
```

I think global function provides better incapsulation.